### DISCUSSION TITLE
Custom Controls for line return wrong position when use toObject to export data

### DISCUSSION THREAD

**Discussion by sohailahmad07** [fabricjs/fabric.js#9952] [Q&A]
Hi i found the code pen online to create custom points for line i work fine expect when you try to export the object with `toObject` the points value is incorrect  

https://codepen.io/Sohail-Ahmad-the-solid/pen/XWwwpgZ

Try move the line by a point and open the console to look the output

**Comment by asturur** [comment]
The issue is the exact same you are having in the other thread around objects in active selection.
The points are not relative to the canvas, are relative to the center of the line.
use the transformPoint utility to multiply the points for the matrix of the line (line.calcTransformMatrix) and you get the point relative to the canvas, if that is what are you looking for.

**Reply by sohailahmad07** [reply]
HI Sir

i do not think so in my other thread #9943 the issue is due to multiple selections when you select multiple objects the position changes and  it is relative to active selection so regardless if use `toObject` or not the object position is incorrect

but here i am selecting only one object one line object its position is correct but when i use `toObject` the position is only wrong then not before


plus i would appreciate if you can give a resource for where i can fix my both issue 


Thanks in advance

**Reply by asturur** [reply]
@sohailahmad07 you continue to open poor quality issues.
`i found some code on the internet but then points are wrong`
you don't say why those points are wrong.
You want a resource but you don't consider this a resource:
```
use the transformPoint utility to multiply the points for the matrix of the line (line.calcTransformMatrix) and you get the point relative to the canvas
```
Then you add this:
```
but here i am selecting only one object one line object its position is correct but when i use toObject the position is only wrong then not before
```
The position is of the points or the line Object? which one?

**Reply by sohailahmad07** [reply]
HI sir sorry for late reply i want the position of the line object 

if you looked at this code pen

 https://codepen.io/Sohail-Ahmad-the-solid/pen/XWwwpgZ

when you try to move the line using the points 

i get the log the position for line object

but the position of  `line.toObject` is incorrect  


When i log the line object

![Screen Capture_select-area_20240701152738](https://github.com/fabricjs/fabric.js/assets/58299055/c7cfe99c-0312-4575-9c80-1874567333f9)


when i log `line.toObject()`

![Screen Capture_select-area_20240701152758](https://github.com/fabricjs/fabric.js/assets/58299055/cb486b90-94e1-4c83-b6b5-e4520a5b6876)



Why are line points incorrect when i use `line.toObject`

**Reply by asturur** [reply]
they are not incorrect again, are different but equivalent.
Those points are relative to the line bounding box center and are good for import/export of the line.
When you reload the line from this data, is the data in the same position? If Yes those points are fine.
The issue is that you didn't manage to write down yet what are you looking for.
Or did you and i missed it?

**Comment by ouyoungs** [comment]
in you case : https://codepen.io/Sohail-Ahmad-the-solid/pen/XWwwpgZ

when move the line first,  then move the line's control point, it incorrect.

is anyone solved this problem?
